ribasim_qgis/core/nodes.py

```diff
@@ -47,102 +49,123 @@
 class Input(abc.ABC):
     """Abstract base class for Ribasim input layers."""
 
-    input_type = ""
-
     def __init__(self, path: str):
-        self.name = self.input_type
-        self.path = path
-        self.layer = None
+        self.name = self.input_type()
```

Comment:
I find this `self.name` a weird concept and I think we can remove it. Since `input_type()` describes exactly this on a class level. Meaning that we could even check up front if we should create the instance or not (line 80).

And for the rest we could simply call input_type() where needed.